pkg/kritis/cryptolib/verifier_test.go

```diff
@@ -73,51 +73,72 @@ const verifierPublicKeyID = "446625EB36036D7546B2D63B77B4BE6989834233"
 
 func TestNewPublicKey(t *testing.T) {
 	tcs := []struct {
-		name        string
-		keyType     KeyType
-		keyData     []byte
-		keyID       string
-		expectedErr bool
-		expectedID  string
+		name               string
+		keyType            KeyType
+		signatureAlgorithm SignatureAlgorithm
+		keyData            []byte
+		keyID              string
+		expectedErr        bool
+		expectedID         string
 	}{
 		{
-			name:        "valid PGP key ID",
-			keyType:     Pgp,
-			keyData:     []byte(verifierPublicKey),
-			keyID:       verifierPublicKeyID,
-			expectedErr: false,
-			expectedID:  verifierPublicKeyID,
+			name:               "valid PGP key ID",
+			keyType:            Pgp,
+			signatureAlgorithm: UndefinedSigningAlgorithm,
+			keyData:            []byte(verifierPublicKey),
+			keyID:              verifierPublicKeyID,
+			expectedErr:        false,
+			expectedID:         verifierPublicKeyID,
 		},
 		{
-			name:        "incorrect PGP key ID",
-			keyType:     Pgp,
-			keyData:     []byte(verifierPublicKey),
-			keyID:       "incorrect-id",
-			expectedErr: false,
-			expectedID:  verifierPublicKeyID,
+			name:               "incorrect PGP key ID",
+			keyType:            Pgp,
+			signatureAlgorithm: UndefinedSigningAlgorithm,
+			keyData:            []byte(verifierPublicKey),
+			keyID:              "incorrect-id",
+			expectedErr:        false,
+			expectedID:         verifierPublicKeyID,
 		},
 		{
-			name:        "valid PKIX key ID",
-			keyType:     Pkix,
-			keyID:       "valid-key-id",
-			expectedErr: false,
-			expectedID:  "valid-key-id",
+			name:               "valid PKIX key ID",
+			keyType:            Pkix,
+			signatureAlgorithm: RsaSignPkcs12048Sha256,
+			keyID:              "valid-key-id",
+			expectedErr:        false,
+			expectedID:         "valid-key-id",
 		},
 		{
-			name:        "invalid PKIX key ID",
-			keyType:     Pkix,
-			keyID:       ":{invalid-key-id}",
-			expectedErr: true,
+			name:               "valid PKIX key ID with undefined signature algorithm",
```

Comment:
nit: make testcase name highlight the error: "PKIX key with incorrect signature algorithm"